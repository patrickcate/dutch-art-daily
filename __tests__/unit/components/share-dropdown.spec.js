import { shallowMount } from '@vue/test-utils'
import ShareDropdown from '@components/share-dropdown.vue'
import SvgIconMock from '@fixtures/svg-icon-mock.vue'

describe('ShareDropdown Component', () => {
  it('should be a Vue instance', () => {
    const wrapper = shallowMount(ShareDropdown, {
      stubs: {
        'svg-icon': SvgIconMock,
      },
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(ShareDropdown, {
      stubs: {
        'svg-icon': SvgIconMock,
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('toggles open and closed', async () => {
    const wrapper = shallowMount(ShareDropdown, {
      stubs: {
        'svg-icon': SvgIconMock,
      },
    })

    expect(wrapper.classes()).not.toContain('share-dropdown--is-expanded')
    expect(wrapper.find('button').attributes('aria-expanded')).toBeUndefined()

    wrapper.find('button').trigger('click')
    await wrapper.vm.$nextTick()

    expect(wrapper.classes()).toContain('share-dropdown--is-expanded')
    expect(wrapper.find('button').attributes('aria-expanded')).toBe('true')
  })
})
