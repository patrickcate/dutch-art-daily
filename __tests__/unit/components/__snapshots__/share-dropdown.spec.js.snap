// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareDropdown Component renders correctly 1`] = `
<div class="share-dropdown"><button id="js-share-dropdown__button" type="button" aria-haspopup="true" class="share-dropdown__button"><svg width="1em" height="1em" class="base-icon icon" role="presentation" focusable="false" aria-hidden="true">
      <use></use>
    </svg> <span class="u-sr-only">Share</span></button>
  <share-list-stub direction="vertical" aria-hidden="true" aria-labelledby="js-share-dropdown__button" class="share-dropdown__list"></share-list-stub>
</div>
`;
